doc/langref.html.in

```diff
@@ -236,20 +238,99 @@ pub fn main() !void {
+      </p>
+      <p>
+        In the <code>hello.zig</code> code sample, the <code>main</code> function is declared
+        with the {#syntax#}!void{#endsyntax#} return type. This return type tells the Zig compiler,
+        and other people reading the code, the function will not return a value and it <i>might</i> fail.
+        The {#syntax#}!{#endsyntax#} (bang, exclamation mark) before the {#syntax#}void{#endsyntax#}
+        {#link|type|Primitive Types#} is what tells the Zig compiler an {#link|error|Errors#} <i>might</i>
+        occur. The {#syntax#}void{#endsyntax#} return type tells the Zig compiler the <code>main</code>
+        function will not return a value.
+      </p>
+      <p>
+        In Zig, a function's block of statements and expressions are surrounded by <code>{</code> and
+        <code>}</code> curly-braces. Inside of the <code>main</code> function are expressions that perform
+        the task of outputting <code>Hello, world!</code> to standard output. 
+      </p>
+      <p>
+        First, a constant identifier, <code>stdout</code>, is initialized to represent the standard output
+        stream. Then, the program tries to print the <code>Hello, world!</code> message to the standard output
+        stream.
+      </p>
+      <p>
+        Functions sometimes need information to perform their task. In Zig, information is passed
+        to functions between open <code>(</code> and close <code>)</code> parenthesis placed after
+        the function's name. The information passed to functions are its arguments. When there are
+        multiple arguments passed to a function, they are separated by commas <code>,</code>.
+      </p>
+      <p>
+        The two arguments passed to the <code>stdout.print()</code> function, <code>"Hello, {}!\n"</code>
+        and <code>.{"world"}</code>, are evaluated at {#link|compile-time|comptime#}. The code sample is
+        purposely written to show how to perform {#link|string|String Literals and Character Literals#}
+        substitution in the <code>print</code> function. The curly-braces inside of the first argument
+        are substituted with the compile-time known value inside of the second argument
+        (known as an {#link|anonymous struct literal|Anonymous Struct Literals#}). The <code>\n</code>
+        inside of the double-quotes of the first argument is the {#link|escape sequence|Escape Sequences#} for the
+        newline character. The {#link|try#} expression evaluates the result of <code>stdout.print</code>.
+        If the result is an error, then the {#syntax#}try{#endsyntax#} expression will return from
+        <code>main</code> with the error. Otherwise, the program will continue. In this case, there are no
+        more statements or expressions left to execute in the <code>main</code> function, so the program exits.
+      </p>
+      <p>
+        In Zig, the standard output stream's <code>print</code> function is allowed to fail because
```

Comment:
In the master branch, `OutStream` is deprecated now. It might make sense to update the documentation and the example code from `outStream` to the new `writer`.

Reply:
Thank you for the notice regarding OutStream. Should the code example update go into another pull request in case this one doesn't get accepted? For coordination, I envision the code example update is merged into ziglang:master first then to this branch then a rewrite to accommodate the changes.

Reply:
@paulespinosa Creating a separate PR for that would also be fine, I guess

Reply:
@joachimschmidt557 I went ahead and made the changes here. I realized later that if there is a separate pull request, then it should merge with these changes cleanly. Thank you.